superbench/benchmarks/micro_benchmarks/directx_gemm_flops_performance/GPUCore.cpp

```diff
@@ -0,0 +1,507 @@
+    // Describe and create the command queue.
+    queueDesc.Flags = D3D12_COMMAND_QUEUE_FLAG_NONE;
+    queueDesc.Type = D3D12_COMMAND_LIST_TYPE_DIRECT;
+
+    D3D12_COMMAND_QUEUE_DESC cqd3 = {};
+    cqd3.Type = D3D12_COMMAND_LIST_TYPE_DIRECT;
+    m_device->CreateCommandQueue(&cqd3, IID_PPV_ARGS(&m_commandQueue));
+
+    m_device->CreateCommandAllocator(D3D12_COMMAND_LIST_TYPE_DIRECT, IID_PPV_ARGS(&m_commandAllocator));
+
+    m_device->CreateCommandList(0, D3D12_COMMAND_LIST_TYPE_DIRECT, m_commandAllocator.Get(), nullptr,
+                                IID_PPV_ARGS(&m_commandList));
+
+    ThrowIfFailed(m_commandList->Close());
+    // Reset the command list to prep for initialization commands.
+    ThrowIfFailed(m_commandList->Reset(m_commandAllocator.Get(), nullptr));
+
+    // Create fence.
+    ThrowIfFailed(m_device->CreateFence(0, D3D12_FENCE_FLAG_NONE, IID_PPV_ARGS(&m_fence)));
+    m_currentFence = 1;
+    // Create an event handle to use for GPU synchronization.
+    m_eventHandle = CreateEvent(0, false, false, 0);
+}
+
+inline UINT64 DMLCalcBufferTensorSize(DML_TENSOR_DATA_TYPE dataType, UINT tensorElementCount) {
+    UINT elementSizeInBytes = 0;
+    switch (dataType) {
+    case DML_TENSOR_DATA_TYPE_FLOAT32:
+
+        elementSizeInBytes = 4;
+        break;
+
+    case DML_TENSOR_DATA_TYPE_FLOAT16:
+
+        elementSizeInBytes = 2;
+        break;
+
+    default:
+        return 0; // Invalid data type
+    }
+
+    UINT64 minimumImpliedSizeInBytes = 0;
+
+    // Round up to the nearest 4 bytes.
+    // Refer to DirectMLX.h
+    minimumImpliedSizeInBytes = (tensorElementCount * elementSizeInBytes + 3) & ~3ull;
+
+    return minimumImpliedSizeInBytes;
+}
+
+/**
+ * @brief Setup and compile DirectML operator.
+ */
+void GPUCore::LoadAssets(int m, int n, int k, DML_TENSOR_DATA_TYPE dataType) {
+    ComPtr<IDMLOperator> dmlOperator;
+
+    // Create DirectML operator(s). Operators represent abstract functions such as "multiply", "reduce",
+    // "convolution", or even compound operations such as recurrent neural nets. This example creates an instance of
+    // the Identity operator, which applies the function f(x) = x for all elements in a tensor.
+    DML_GEMM_OPERATOR_DESC dmlGEMMOperatorDesc{};
+
+    DML_TENSOR_DESC dmlTensorDescA{};
+    dmlTensorDescA.Type = DML_TENSOR_TYPE_BUFFER;
+
+    DML_BUFFER_TENSOR_DESC bindingDescA = {};
+    UINT tensorSizesA[4] = {1, 1, static_cast<UINT>(m), static_cast<UINT>(k)};
+    UINT tensorElementCount = tensorSizesA[0] * tensorSizesA[1] * tensorSizesA[2] * tensorSizesA[3];
+    bindingDescA.DataType = dataType;
+    bindingDescA.Flags = DML_TENSOR_FLAG_NONE;
+    bindingDescA.DimensionCount = ARRAYSIZE(tensorSizesA);
+    bindingDescA.Sizes = tensorSizesA;
+    bindingDescA.Strides = nullptr;
+    bindingDescA.TotalTensorSizeInBytes = DMLCalcBufferTensorSize(dataType, tensorElementCount);
+
+    dmlTensorDescA.Desc = &bindingDescA;
+    dmlGEMMOperatorDesc.ATensor = &dmlTensorDescA;
+
```

Comment:
looks like you repeat the similar logic multiple times, so how about create a function to do that?

Reply:
created CreateTensorDesc function to do that, please help check